stl/inc/ranges

```diff
@@ -7656,6 +7662,401 @@ namespace ranges {
+            : _Function(in_place, _STD move(_Function_)), _Zip(_STD move(_Views)...) {}
+
+        constexpr auto begin() noexcept(noexcept(_Iterator<false>{*this, _Zip.begin()})) // strengthened
+        {
+            return _Iterator<false>{*this, _Zip.begin()};
+        }
+
+        constexpr auto begin() const noexcept(noexcept(_Iterator<true>{*this, _Zip.begin()})) // strengthened
+            requires _Enable_const_begin_end
+        {
+            return _Iterator<true>{*this, _Zip.begin()};
+        }
+
+        constexpr auto end() noexcept(_Is_end_noexcept<false>) // strengthened
+        {
+            if constexpr (common_range<_Inner_view>) {
+                return _Iterator<false>{*this, _Zip.end()};
+            } else {
+                return _Sentinel<false>{_Zip.end()};
+            }
+        }
+
+        constexpr auto end() const noexcept(_Is_end_noexcept<true>) // strengthened
+            requires _Enable_const_begin_end
+        {
+            if constexpr (common_range<const _Inner_view>) {
+                return _Iterator<true>{*this, _Zip.end()};
+            } else {
+                return _Sentinel<true>{_Zip.end()};
+            }
+        }
+
+        constexpr auto size() noexcept(noexcept(_Zip.size())) // strengthened
+            requires sized_range<_Inner_view>
+        {
+            return _Zip.size();
+        }
+
+        constexpr auto size() const noexcept(noexcept(_Zip.size())) // strengthened
+            requires sized_range<const _Inner_view>
+        {
+            return _Zip.size();
+        }
+    };
+
+    template <class _Func, class... _Ranges>
+    zip_transform_view(_Func, _Ranges&&...) -> zip_transform_view<_Func, views::all_t<_Ranges>...>;
+
+    namespace views {
+        struct _Zip_transform_fn {
+        private:
+            template <class _Func, class... _Types>
+            static constexpr bool _Is_invocation_noexcept = []() noexcept {
+                if constexpr (sizeof...(_Types) == 0) {
+                    // equivalent to:
+                    // using _Decayed_func = decay_t<_Func>;
+                    //
+                    // return noexcept(((void) _STD declval<_Func&>(),
+                    //     auto(empty<decay_t<invoke_result_t<_Decayed_func&>>>)));
+                    return true;
+                } else {
+                    return noexcept(zip_transform_view(_STD declval<_Func>(), _STD declval<_Types>()...));
+                }
+            }();
+
+            template <class _Func, class... _Types>
+            static constexpr bool _Enable_cpo = []() noexcept {
+                if constexpr (sizeof...(_Types) == 0) {
+                    using _Decayed_func = decay_t<_Func>;
+
+                    if constexpr (move_constructible<_Decayed_func> && regular_invocable<_Decayed_func&>) {
+                        return !is_void_v<invoke_result_t<_Decayed_func&>>;

```

Comment:
This is a correct transformation of the Standardese, but @CaseyCarter and I had to think for a moment to understand what was happening. It would be good to comment this by saying that it implements N4928 [range.zip.transform.overview]/2.1.1 more efficiently.